client/index.jsx

```diff
@@ -140,17 +142,19 @@ Promise.all([rootElemPromise])
   .then(({ elem, store }) => {
     const root = createRoot(elem)
     root.render(
-      <RootErrorBoundary>
-        <ReduxProvider store={store}>
-          <Router>
-            <RedirectProvider>
-              <>
-                <App />
-                {ReduxDevToolsContainer ? <ReduxDevToolsContainer /> : null}
-              </>
-            </RedirectProvider>
-          </Router>
-        </ReduxProvider>
-      </RootErrorBoundary>,
+      <React.Suspense fallback={<LoadingIndicator />}>
```

Comment:
This should at the very least be inside the error boundary imo, but you could probably put it deeper than that (e.g. wrap the App directly)

Reply:
My thinking was that we'd want the error boundary translated as well, so I wasn't sure if this Suspense needs to be above it for some reason. But probably not now that I think about it.

Reply:
This wouldn't really make that happen, the suspense doesn't cause anything to be available or not, it just catches the thrown promises and shows a loading indicator. The point of the root error boundary is that it will catch *all* errors, so that nothing ends up totally crashing the app, so it should be outside of everything. (Since, potentially, there could be errors in the translation loading stuff, we don't want to let those pass through uncaught).

I would agree we'd want to translate that stuff if possible, but you need to handle the case that an error happened before they got loaded regardless

Reply:
Also just for future reference, if you *do* want to translate strings inside the RootErrorBoundary UI, you need to do so via something that *does not* utilize Suspense/throw promises/etc. It would need to try to use translations if they're available immediately but fall back to the defaults if not, because once the RootErrorBoundary is hit we can no longer trust that any app state is correct, that fetches will work, etc.

Reply:
Yeah, that's a good point. I'll try to keep it in mind :d

Another thing I was wondering about Suspense that I'm not sure how it works. For example, when the user switches the language, the new translation file is being fetched and the Suspense renders the fallback while that happens. But, does the user loses all transient state when the fallback is rendered (e.g. by unmounting all the component below the Suspense)?

Reply:
I'm not sure, I haven't used Suspense extensively. It should be easy to write a small demo that would tell you, though. (I also don't think it's really a big deal were that the case, I'd just change the flow such that language changes notify users they must reload the app, then do so.

Also, after thinking about this more, this Suspense actually needs to be even deeper in the tree, it doesn't make sense to hide the window controls just because translation files are loading